HouseRules_Essentials/Rules/TileEffectDurationOverriddenRule.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace HouseRules.Essentials.Rules
+{
+    using System.Collections.Generic;
+    using Boardgame;
+    using Boardgame.Board;
+    using DataKeys;
+    using HarmonyLib;
+    using HouseRules.Types;
+
+    public sealed class TileEffectDurationOverriddenRule : Rule, IConfigWritable<Dictionary<TileEffect, int>>, IPatchable, IMultiplayerSafe
+    {
+        public override string Description => "Tile Effect durations are overridden.";
+
+        private static Dictionary<TileEffect, int> _globalAdjustments;
+        private static bool _isActivated;
+
+        private readonly Dictionary<TileEffect, int> _adjustments;
+
+
+        public TileEffectDurationOverriddenRule(Dictionary<TileEffect, int> adjustments)
+        {
+            _adjustments = adjustments;
+        }
+
+        public Dictionary<TileEffect, int> GetConfigObject() => _adjustments;
+
+        protected override void OnActivate(GameContext gameContext)
+        {
+            _globalAdjustments = _adjustments;
+            _isActivated = true;
+        }
+
+        protected override void OnDeactivate(GameContext gameContext) => _isActivated = false;
+
+        private static void Patch(Harmony harmony)
+        {
+            harmony.Patch(
+                original: AccessTools.Method(typeof(TileEffectState), "GetLayerExpireTime"),
+                prefix: new HarmonyMethod(
+                    typeof(TileEffectDurationOverriddenRule),
+                    nameof(TileEffectState_GetLayerExpireTime_Prefix)));
+        }
+
+        private static bool TileEffectState_GetLayerExpireTime_Prefix(ref TileEffect type, ref int __result)
```

Comment:
nit: could drop `ref` from `TileEffect` since the ref feature goes unused.

Reply:
Without the ref, it was sad because I wasn't passing the 'type' parameter into the function. Ref seemed to make it all happy again.